java/code/src/com/redhat/rhn/taskomatic/task/repomd/DebReleaseWriter.java

```diff
@@ -99,7 +100,10 @@ public void generateRelease() {
     }
 
     private String toArchString(ChannelArch channelArch) {
-        return channelArch.getLabel().replaceAll("channel-", "").replaceAll("-deb", "");
+        return (String)channelArch.getCompatiblePackageArches().stream()
+                .map(a -> ((PackageArch)a).getLabel().replaceAll("-deb", ""))
+                .filter(a -> !("all".equals(a) || "src".equals(a)))
+                .collect(Collectors.joining(" "));
```

Comment:
Let's add `sorted()` to adjust the test:
```suggestion
                sorted().collect(Collectors.joining(" "));
```